apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/TransactionalDataSource.java

```diff
@@ -61,6 +61,10 @@
     private volatile long txTimes = 0;
     private volatile long txCount = 0;
     private volatile long statsTime = 0;
+    /**
+     * Optional.EMPTY for 'main db', Optional.value 1...xx for shardId, Optional.value NEGATIVE = -1 for temp db only
+     */
+    private Optional<Long> dbIdentity = Optional.empty();
```

Comment:
We can change it as we need, now logic is described above - 
**Optional.EMPTY for 'main db', Optional.value 1...xx for shardId,....**
So we have 'sign of main db' in most cases, but 'shard db sign' is needed for sharding process.

Reply:
I think, that enum will increase readability of the code

Reply:
I don't need ENUM because I need 'shardId value' for steps of shard creation process.
I need to keep shardId between two shard db schema updates. So I decided to put that information directly into every TransactionalDataSource.java, but leave it Optional value, because it rarely used.

Reply:
Ok